### ISSUE TITLE
Explicit content filter not working sometimes

### ISSUE THREAD

**Issue by meyertime** [pithos/pithos#623]
Hi, first of all thanks for this Linux Pandora client!  I'm a newish Linux desktop user (past 1.5 years) but long time Pandora user (~10 years).  I've been using Pithos to help me focus while I write code at work.

I have noticed, however, that the explicit content filter does not work correctly in Pithos compared to the web player which I always used previously.  Just to give a little background, I've always had the explicit content filter on in Pandora just because I find cursing distasteful.

Generally, explicit songs don't come up on my stations in either Pithos or any other player.  However, there are cases where there are songs with some cursing which is edited when played from the web, but in Pithos, the explicit version plays.  (One example is "Feel Good Inc." by Gorillaz)  Please excuse my ignorance, because I don't know anything about the Pandora API, but I'm wondering if the playlist calls return multiple versions of a song, some of which may be explicit while others are edited, and the player may have to choose the edited version when the explicit content filter is on.

Just a thought.  Maybe if you can provide any insight into what might be happening and could point me in the right direction, I could take at look at the code too.  Thanks!

**Comment by TingPing** [comment]
Pithos sets the explicit filter on your Pandora account basically; it has no control over what playlists Pandora returns.

**Comment by meyertime** [comment]
@TingPing Thanks.  I understand that part.  But Pithos still behaves differently than the web player, so there must be more to it.

In this case, the explicit song may still be returned by Pandora in the playlist because there is an edited clean version available that the web player plays, but Pithos streams the explicit version.

**Comment by meyertime** [comment]
So I guess it was just coincidence that I started using Pithos at the same time that the explicit filter stopped working correctly.  I've been using the web player for a while just to verify that it behaves differently, and the same song "Feel Good Inc." just came on, and it also played uncensored.  So apparently it's an issue with Pandora.  I tried turning the explicit filter off and on again, so we'll see if that helps.

I will close the issue but comment here if I end up finding a solution.

I also found that others are experiencing this issue with Pandora: https://community.pandora.com/t5/My-Collection/Explicit-Content-filter-is-not-working/td-p/2791
